File: Swift/프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.swift
import Foundation

func solution(_ my_string:String, _ overwrite_string:String, _ s:Int) -> String {
    let prefix = my_string.prefix(s)
    let suffix = my_string.dropFirst(s + overwrite_string.count)
    return prefix + overwrite_string + suffix
}


import Foundation

func solution(_ my_string: String, _ overwrite_string: String, _ s: Int) -> String {
    var result = my_string
    let startIndex = my_string.index(my_string.startIndex, offsetBy: s)
    let endIndex = my_string.index(startIndex, offsetBy: overwrite_string.count)
    
    result.replaceSubrange(startIndex..<endIndex, with: overwrite_string)
    
    return result
}

File: Java/프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.java
import java.util.*;

public class Solution { 
    public String solution (String my_string, String overwrite_string, int s) {
        
        StringBuilder result = new StringBuilder(my_string);
        
        result.replace(s, s + overwrite_string.length(), overwrite_string);
        
        return result.toString();
    }    
}